component/src/main/java/org/cbioportal/genome_nexus/util/IsoformOverrideSource.java

```diff
@@ -0,0 +1,19 @@
+package org.cbioportal.genome_nexus.util;
+
+public class IsoformOverrideSource {
+    public static final String ENSEMBL = "ensembl";
+    public static final String GENOME_NEXUS = "genome_nexus";
+    public static final String UNIPROT = "uniprot";
+    public static final String MSKCC = "mskcc";
+
+    public static String getOrDefault(String isoformOverrideSource) {
+        String normalizedSource = isoformOverrideSource == null ?
+            UNIPROT : isoformOverrideSource.trim().toLowerCase();
+
+        return (
+            normalizedSource.equals(ENSEMBL) ||
+            normalizedSource.equals(GENOME_NEXUS) ||
+            normalizedSource.equals(MSKCC)
+        ) ? normalizedSource : UNIPROT;
```

Comment:
does this mean if someone passes in something "fake_source" we'll still be using UNIPROT?

Reply:
I see, looks like that was basically existing behavior (At least in EnsemblCanonical class)

Reply:
right, currently uniprot is the default.